core/src/debug_client.rs

```diff
@@ -0,0 +1,71 @@
+//! This module contains code for an http client that is used for the VSCode debug plugin.
+
+use prost::Message;
+use reqwest;
+use std::time::Duration;
+use temporal_sdk_core_protos::temporal::api::history::v1::History;
+
+const CLIENT_NAME: &str = "temporal-core";
+const CLIENT_VERSION: &str = "0.1.0";
+
+/// Struct representing the internal debug client
+#[derive(Clone)]
+pub struct DebugClient {
+    /// URL for the local instance of the debugger server
+    debugger_url: String,
+
+    /// reqwest::Client to avoid re-creating a client for every request.
+    client: reqwest::Client,
+}
+
+impl DebugClient {
+    /// Create a new instance of a DebugClient with the specified url.
+    pub fn new(url: String) -> DebugClient {
+        DebugClient {
+            debugger_url: url,
+            client: reqwest::Client::new(),
+        }
+    }
+
+    /// Get the history from the instance of the debug plugin server, then return the protobinary if successful.
```

Comment:
This isn't actually encoded yet, so you can just lop that bit off
```suggestion
    /// Get the history from the instance of the debug plugin server
```